kernel/src/scan/mod.rs

```diff
@@ -103,9 +108,63 @@ pub struct ScanResult {
 
 /// Scan uses this to set up what kinds of columns it is scanning. A reference to the field is
 /// stored so we can get the name and data_type out when building the read expression
-enum ColumnType<'a> {
-    Selected(&'a StructField),
-    Partition(&'a StructField),
+pub enum ColumnType<'a> {
+    Selected(Cow<'a, StructField>),
+    Partition(Cow<'a, StructField>),
+}
+
+impl<'a> ColumnType<'a> {
+    pub(crate) fn into_owned(self) -> ColumnType<'static> {
+        match self {
+            ColumnType::Selected(field) => {
+                let field = field.into_owned();
+                ColumnType::Selected(Cow::Owned(field))
+            }
+            ColumnType::Partition(field) => {
+                let field = field.into_owned();
+                ColumnType::Partition(Cow::Owned(field))
+            }
+        }
+    }
+}
+/// One file to scan, plus associated metadata
+pub struct ScanFile {
+    table_root: Url,
+    path: String,
+    pub size: usize,
+    pub partition_values: std::collections::HashMap<String, String>,
+    deletion_vector_descriptor: Option<DeletionVectorDescriptor>,
+}
+
+impl ScanFile {
```

Comment:
Well, this is a design decision. 

Some of the data should just be forwarded blindly back, but there are other things where you need to poke at the data. In the example above I [note](https://github.com/delta-incubator/delta-kernel-rs/pull/169/files#diff-3f44d0b7f8cfbe763cbe0cdbb2e2450a84833065de32fc102aef9d38b21b3daaR203) that we could push the selection vector (deletion vector) down into the parquet scanner if desired, which requires more than just opaque data. 

And of course the data returned from `location()` below is used by the engine to read the raw parquet. We _could_ require the engine to round trip the data into the parquet handler, but I know for example already that the duckdb folks prefer to just get a raw path to the file.

Reply:
Note: 
> [jvm kernel] defines extractor methods that engine can use to get necessary details like file names

^^ would _seem_ to satisfy both the requirements you mention (quoted below)?

> [pushing] the selection vector (deletion vector) down into the parquet scanner if desired, which requires more than just opaque data.


> the duckdb folks prefer to just get a raw path to the file.

Reply:
Yeah okay. So I think there's a couple of different decisions in here. 

One option:
1. "JVM kernel just passes generic rows back to engine" -> This means we'd just apply some expression to the `engine_data` containing the `Add`s, to transform it into a new `engine_data` with the correct info?
2. If yes to above, seems like we wouldn't need extractors because the data would just be in engine format already?
3. This implies though that the engine needs to be able to send its own data between threads/nodes. Which is probably fine, but is an extra requirement on the engine. In the examples in the jvm they have to define `RowSerDe.serializeRowToJson` and `RowSerDe.deserializeRowFromJson` for this reason.

Another option:
1. Don't make any of the fields `pub` here, and _only_ provide extractor methods like `location()`. (Although I'm not 100% sure that's what you mean by an "extractor" method.
2. Then we _do_ have to provide our own ser/de for this struct, but we can make that be anything we want.
3. This then frees the engine from having to think about ser/de.

Reply:
Thing is, ser/de is an engine concern. Kernel shouldn't be in the business of sending things over the wire or to disk.

Further, the engine will already have a ser/de answer for its own engine data. Either it's not a distributed engine (so the engine data pointer can be passed around directly), or it knows how to serialize the engine data when passing results from executors back to driver. 

(we had a very similar discussion with jvm kernel several months ago)

As far as extractors go, a middle ground is to publish a subset of the schema. E.g. engine knows the engine data that comes back from log replay will contain fields like `add.path`. I don't remember the details of how kernel jvm ended up implementing it.